src/components/polygons/boundary-drawer.vue

```diff
@@ -216,27 +234,55 @@ export default class BoundaryDrawer extends Vue {
     }
 
     private get transformRoot(): SVGGraphicsElement {
-        return this.svg.getElementById(this.transformRootId) as SVGGraphicsElement;
+        return this.svg.getElementById(
+            this.transformRootId
+        ) as SVGGraphicsElement;
     }
 
     @Emit()
-    private newPolygon() {
+    private newPolygon(polygon: Polygon) {
+        return polygon;
+    }
+
+    private async checkPolygon() {
         // Turn the polygon into an svg string, due it bit by bit
-        const partials = [`M${this.polygonPoints[0].x} ${this.polygonPoints[0].y}`];
+        const partials = [
```

Comment:
It seems to me that this construction of an SVG should be in a service somewhere (maybe it even already exists somewhere), since converting these polygon types probably can happen in several places in the app.